cli/src/oauth/mod.rs

```diff
@@ -19,17 +20,31 @@ const CODE_BODY: &'static str = "{\
 const POLL_URL: &'static str = "https://github.com/login/oauth/access_token";
 const ERR: &'static str = "Failed to perform OAuth 2.0 authentication with GitHub";
 const TOKEN_FILE: &str = ".anycloud/.token";
+static TOKEN: OnceCell<String> = OnceCell::new();
 
-// Get previously generated OAuth access token or generate a new one
-pub async fn get_token() -> String {
-  let home = std::env::var("HOME").unwrap();
-  let file_name = &format!("{}/{}", home, TOKEN_FILE);
-  match read_to_string(file_name) {
-    Ok(token) => token,
-    Err(_) => generate_token().await,
+// Get saved token
+pub fn get_token() -> &'static str {
+  let token = TOKEN.get();
+  if let Some(token) = token {
+    return token;
+  } else {
+    return "";
```

Comment:
when does this happen?

Reply:
When failed to authenticate

Reply:
this should probably be something like
```rs
return token.expect("No token found");
```

Reply:
let's just do `return token.expect(ERR)` like we do in authenticate

Reply:
Though we wouldn't get the error message for that, since we depend on the token for logging purposes :/

Perhaps this should also be an `async` function and we have a special log sent just for this condition if ever hit?

Reply:
Hmm the issue is that when this happen we still want to send the empty token to the deploy service to log the event of authentication fail

Reply:
> Though we wouldn't get the error message for that, since we depend on the token for logging purposes :/
> 
> Perhaps this should also be an `async` function and we have a special log sent just for this condition if ever hit?

Yes! @aguillenv mentioned the `clientError` endpoint will work without an access token only for an authentication error, right?

Reply:
I'm guessing that would be part of https://github.com/alantech/anycloud/pull/41 though

Reply:
Yes, it would, but I cannot `.expect`, I need to send the empty token so the request can continue. On the deploy service side, will catch error and empty token to let it log the event.

Reply:
Can we leave a comment about that and why we let it be empty when fails to authenticate?